### PR TITLE
Update datasets spicepod spec

### PR DESCRIPTION
**Pull request by ewgenius**

### CODE REVIEW

File: spiceaidocs/content/en/reference/Spicepod/datasets.md [hunk 1]
```diff
@@ -44,63 +44,55 @@ datasets:
...
-- `append` - Appends new data from dataset source to the dataset.
+Where:
+
+- `<source>`: The source of the dataset
 
-## `source`
+  Currently supported sources:
+  - `spiceai`
+  - `dremio`
+  - `databricks`
```
Comment (y-f-u) [inline spiceaidocs/content/en/reference/Spicepod/datasets.md:82]: s3 too.

Comment (y-f-u) [reply spiceaidocs/content/en/reference/Spicepod/datasets.md:82]: flightsql could be there in time, @Jeadie can update here

Comment (ewgenius) [reply spiceaidocs/content/en/reference/Spicepod/datasets.md:82]: Synced with available sources from https://github.com/spiceai/spiceai?tab=readme-ov-file#supported-data-connectors

File: spiceaidocs/content/en/reference/Spicepod/datasets.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Inline example:
 `spicepod.yaml`
 ```yaml
 datasets:
-  - from: spice.ai/eth/beacon/eigenlayer
+  - from: spiceai:spice.ai/eth/beacon/eigenlayer
```
Comment (lukekim) [inline spiceaidocs/content/en/reference/Spicepod/datasets.md]: `spiceai:` is not required (it's special, we default to it).

File: spiceaidocs/content/en/reference/Spicepod/datasets.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ datasets:
 `spicepod.yaml`
 ```yaml
 datasets:
-  - from: databricks.com/spiceai/datasets
+  - from: databricks:databricks.com/spiceai/datasets
```
Comment (Jeadie) [inline spiceaidocs/content/en/reference/Spicepod/datasets.md]:
```suggestion
  - from: databricks:spiceai.datasets.specific_table
```

File: spiceaidocs/content/en/reference/Spicepod/datasets.md [hunk 4]
```diff
@@ -44,63 +44,57 @@ datasets:
...
+
+- `<source>`: The source of the dataset
 
-## `source`
+  Currently supported sources:
+  - `spiceai`
+  - `dremio`
+  - `databricks`
+  - `s3`
+  - `postgres`
```
Comment (lukekim) [inline spiceaidocs/content/en/reference/Spicepod/datasets.md:84]: Add comment noting it will default to `spiceai:`

**Review by mitchdevenport — APPROVED** [review]

**Pull request merged**